Parse/Internal/PFDevice.m

```diff
@@ -27,14 +27,19 @@
     const char *charName = [name UTF8String];
 
     size_t size;
-    sysctlbyname(charName, NULL, &size, NULL, 0);
+    if (sysctlbyname(charName, NULL, &size, NULL, 0) < 0) {
+      return nil;
+    }
     char *answer = (char*)malloc(size);
 
     if (answer == NULL) {
         return nil;
     }
 
-    sysctlbyname(charName, answer, &size, NULL, 0);
+    if (sysctlbyname(charName, answer, &size, NULL, 0) < 0) {
+      free(answer);
+      return nil;
+    }
     NSString *string = [NSString stringWithUTF8String:answer];
```

Comment:
If we're truly being safe here, we should manually pass the length of the bytes here. There's nothing in the `sysctl` man page that indicates that the strings returned through it should be NUL-terminated, even if they are in practice.